setuptools/command/sdist.py

```diff
@@ -198,3 +199,27 @@ def read_manifest(self):
                 continue
             self.filelist.append(line)
         manifest.close()
+
+    def check_license(self):
+        """Read the setup configuration file ('setup.cfg') and use it to find
+        if a license is defined with the 'license_file' attribute.
+        If the license is declared and exists, it will be added to
+        'self.filelist'.
+        """
+
+        cfg_file = 'setup.cfg'
+        log.debug("Reading configuration from %s", cfg_file)
+        parser = configparser.RawConfigParser()
```

Comment:
It's a little disconcerting to me that this routine needs to parse the setup.cfg file. I would expect that whatever logic is needed to include this file by default would happen _after_ the setup.cfg is parsed and the 'license_file' is set as an attribute on some class (Distribution, maybe?). It shouldn't be necessary to re-read the configuration file... or if it is, it should re-use the same code.

Did you or could you explore such an approach?

Reply:
I initially tried `setuptools.config.read_configuration('setup.cfg')`, which does not parse `license_file` from `[metadata]` even if it exists. (Is this expected?)

After your comment, I did manage to find a way to get this without parsing `setup.cfg` again (see 4e4efa77)